File: lib/easy/can_make_arithmetic_progression.dart
bool canMakeArithmeticProgression(List<int> arr) {
  arr.sort((a, b) => a.compareTo(b));
  print(arr);
  int distance = arr[0] - arr[1];
  int index = 1;
  while (index < arr.length - 1) {
    if (distance != arr[index] - arr[index + 1]) return false;
    index++;
  }
  return true;
}

File: src/main/php/Easy/CanMakeArithmeticProgression.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class CanMakeArithmeticProgression
{
    /**
     * @param  int[]  $arr
     */
    public function solve(array $arr): bool
    {
        $nums = $arr;
        sort($nums);
        $index = 0;
        $distance = $nums[1] - $nums[0];
        while ($index < count($nums) - 1) {
            if ($distance != $nums[$index + 1] - $nums[$index]) {
                return false;
            }
            $index++;
        }

        return true;
    }
}
